src/nupic/algorithms/TemporalMemory.cpp

```diff
@@ -508,9 +510,23 @@ void TemporalMemory::activateCells(size_t activeColumnsSize,
       }
     }
   }
+  if( learn )
+    iteration_++;
 }
 
-void TemporalMemory::activateDendrites(bool learn) {
+void TemporalMemory::activateDendrites(bool learn,
+                                       const vector<UInt> &extraActive,
+                                       const vector<UInt> &extraWinners) {
+
+  for(const auto &active : extraActive) {
+    NTA_ASSERT( active < extra_ );
+    activeCells_.push_back( active + numberOfCells() );
```

Comment:
add somewhere doc like: "Ranges for cell indices: |0..<= normal cells activation < numberOfCells | <= xxx activation <  extraActive| <= winners ... < extraWinners|" 
...or something along this meaning